test/browser/component.js

```diff
@@ -126,6 +126,25 @@ describe(`Simple Component instance`, function() {
     });
   });
 
+  context(`when detached from DOM`, function() {
+    beforeEach(async function() {
+      document.body.appendChild(el);
+      await nextAnimationFrame();
+      document.body.removeChild(el);
+      await nextAnimationFrame();
+    });
+
+    it(`disconnectedCallback cleans up to be GC friendly`, function() {
+      expect(el.$panelRoot).to.equal(null);
+      expect(el.$panelParent).to.equal(null);
+      expect(el.appState).to.equal(null);
+      expect(el.app).to.equal(null);
+      expect(el.domPatcher).to.equal(null);
+      expect(el._rendered).to.equal(null);
+      expect(el.initialized).to.equal(false);
+    });
```

Comment:
Let's get another test case here exercising the path of a component getting removed and reattached (maybe a couple times?) to ensure that the cleanup and reinit cycle is clean

Reply:
done.